### PR TITLE
Refresh order list with order status selected filters

### PR DESCRIPTION
**Pull request by JorgeMucientes**
Work in progress: #4934

### Description
Part II of the order status filters. In this PR you'll see the changes needed to apply the selected order status filters to the order list. 

In Part I #5039 the UI for selecting order status filters was added. 

One important consideration. In order to be able to merge this PR as well as Part I #5039, changes are protected by feature flag to only display the filters button on debug build. This has some implications to be aware of: 
- Order list tabs have been kept
- All the code to load orders for the different tabs has been kept
- Code for searching orders by order status has also been kept

This means that right now there is a lot of code from `OrderListViewModel` that has to be removed in following PRs, when the feature is ready to be released. This code right now conflicts with the filtering code, so beware the the following "misbehavior" is to be expected at the moment: 
- When filtering order list and then navigating back and forth to order list with filters selected, the filtered list will be lost even though the UI shows filters as selected.  The "processing" orders list will be shown as if we were in the "Processing tab" although we can't see it.
- When navigating back to order list after selecting filters, race conditions may occur between the code that loads the orders for the 2 tabs and the code that loads the filtered order list. This may result into an error. Not to be worried about as code related to loading orders for tabs will be removed and ViewModel will be refactored to load the order list once with the corresponding selected filters.  

### Testing instructions

- Compile debug version
- Open orders list
- Click on filters button
- Select some order status filers
- Go back to order list and see how filters are applied.

### Images/gif

https://user-images.githubusercontent.com/2663464/139015332-ebd2cd60-a999-4520-8375-30bf5683c98e.mp4

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/filters/data/OrderFiltersRepository.kt [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package com.woocommerce.android.ui.orders.filters.data
+
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class OrderFiltersRepository @Inject constructor() {
```
Comment (JorgeMucientes) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/filters/data/OrderFiltersRepository.kt:7]: In the next iteration this repository will persist on disk the selected values

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/orders/OrderListViewModelTest.kt [hunk 2]
```diff
@@ -1,21 +1,14 @@
 package com.woocommerce.android.ui.orders
```
Comment (JorgeMucientes) [inline WooCommerce/src/test/kotlin/com/woocommerce/android/ui/orders/OrderListViewModelTest.kt:1]: You can ignore changes to this file. Just renamed repository field

**Review by anitaa1990 — COMMENTED** [review]
Nice work @JorgeMucientes 🏅 The code changes look good! 
I faced some minor issues when testing:
- Filter is not applied when clicking on an order and coming back to the order list. This also happens when I rotate the device when I am on the order list screen (after filters are applied). This only happens when we enable `Do not keep activities` though.


- I believe the filter needs to be applied, if filters are selected and user searches for something 😅 ? That is how the product filter option works at the moment. Do we want to follow a similar logic here?
- When filters are applied to the order list, do we want to change the name of the heading from `All Orders` -> `Filtered Orders`? I remember some discussion on changing the heading based on the filters selected.
- This screen popped up somewhere during testing when clicking on the search icon. I assume this will be removed when we are ready to release this feature 😅  so I’ll ignore this for now!

**Comment by JorgeMucientes** [comment]
@anitaa1990 thank you for the careful testing and highlighting the different issues 🙂

>I faced some minor issues when testing:

Not minor at all 😆  It's all broken, but kind of expected at this phase.  Basically, right now filters are only applied when you come back from the filters screen, in any other scenario where order list is loaded/refreshed the filters will be ignored when fetching data. This is because `OrderListViewModel` requires a big refactor on how data is fetched. But this refactor will break the current 2 tabs UI, and we want to keep that working in order to be able to merge this smaller tasks.  I will leave that refactor to the end of the project where the tabs code will be removed, and then I'll make sure any persisted filters are applied whenever order list is fetched. 
Also the search feature won't show any order status options after the refactor. It will only be text based search. But for the same reason we need to keep this for now. 
I know it's kind of annoying to test thing like this 😅  

As for this: 
>When filters are applied to the order list, do we want to change the name of the heading from All Orders

You are absolutely right! Good catch! I will address that in this PR and ping you once it's pushed.

**Comment by JorgeMucientes** [comment]
Ready to review that last part @anitaa1990 I completely forgot to ping you last Friday 🤦🏼 

Anyway, there is no rush in merging this.

**Review by anitaa1990 — APPROVED** [review]
> Not minor at all 😆 It's all broken, but kind of expected at this phase. Basically, right now filters are only applied when you come back from the filters screen, in any other scenario where order list is loaded/refreshed the filters will be ignored when fetching data. This is because OrderListViewModel requires a big refactor on how data is fetched. But this refactor will break the current 2 tabs UI, and we want to keep that working in order to be able to merge this smaller tasks. I will leave that refactor to the end of the project where the tabs code will be removed, and then I'll make sure any persisted filters are applied whenever order list is fetched.
Also the search feature won't show any order status options after the refactor. It will only be text based search. But for the same reason we need to keep this for now.

This makes complete sense! Thanks for the context @JorgeMucientes 😄 Everything else works great 👏 I'll merge this PR now :shipit:

**Pull request merged**